x/wasm/keeper/gas_register.go

```diff
@@ -144,7 +145,7 @@ func (g WasmGasRegister) ReplyCosts(pinned bool, reply wasmvmtypes.Reply) sdk.Ga
 		msgLen += len(reply.Result.Ok.Data)
 		var attrs []wasmvmtypes.EventAttribute
 		for _, e := range reply.Result.Ok.Events {
-			msgLen += len(e.Type)
+			eventGas += sdk.Gas(len(e.Type)) * g.c.EventAttributeDataCost
```

Comment:
I think this better expresses the intend because the factor is: _gas per byte_

```suggestion
			eventGas += sdk.Gas(len(e.Type) * g.c.EventAttributeDataCost)
```

Reply:
Oh boy, those types. Multiplying gas with gas makes no sense at all but is done everywhere. I give up here.

Reply:
It's not about intent. It is about forcing the GO compiler.

sdk.Gas is an alias for `uint64`. `len()` returns `int`. You cannot multiply `int` with `uint64` without a cast first. Thus I had to cast it before multiplying.